File: src/main/kotlin/application/model/search/cfind/ShipSearchParameters.java
package application.model.search.cfind;

import application.model.rules.Rule;
import application.model.search.SearchParameters;

/**
 * The parameters for CAViewer's ship search program - cfind.
 */
public class ShipSearchParameters extends SearchParameters {
    private final Rule rule;
    private final int dx, dy, period;

    public ShipSearchParameters(Rule rule, int dx, int dy, int period) {
        this.rule = rule;
        this.dx = dx;
        this.dy = dy;
        this.period = period;
    }

    public Rule getRule() {
        return rule;
    }

    public int getDx() {
        return dx;
    }

    public int getDy() {
        return dy;
    }

    public int getPeriod() {
        return period;
    }
}

File: src/main/kotlin/application/model/search/cfind/ShipSearchParameters.kt
package application.model.search.cfind

import application.model.rules.Rule
import application.model.search.SearchParameters
import kotlin.math.pow

data class ShipSearchParameters(val rule: Rule, val width: Int, val dy: Int, val period: Int,
                                val symmetry: Symmetry = Symmetry.ASYMMETRIC,
                                val maxQueueSize: Long = 2.0.pow(20).toLong(),
                                val minDeepingIncrement: Int = period,
                                var lookupTableWidth: Int = 4,
                                var lookahead: Boolean = true, val stdin: Boolean = false,
                                val noPartials: Boolean = false, val dfs: Boolean = false,
                                val randomSearchOrder: Boolean = false): SearchParameters()